docs/developer/development-environment.md

```diff
@@ -0,0 +1,130 @@
+# Configuring Development Environment
+
+This article is written to guide developers to create development environment for developers. This guide can also be used to build Fed-BioMed software from its source.
+
+
+This article will guide you to:
+
+- clone Fed-BioMed source repository.
+- setup a recommended tool to manage different Python versions.
+- setup some optional tools to manage isolated python environments such as `conda` and `virtualenv`.
+- setup pdm to manage dependencies and build operations, and it usage.
+
+
+## Clone the repository
+The first step is to clone the repository please go to (Fed-BioMed GitHub Repository)[https://github.com/fedbiomed/fedbiomed] and clone it your local environment.
+
+```
+git clone https://github.com/fedbiomed/fedbiomed <path-to-clone>
+cd <path-to-clone>
+```
+
+
+## Compatible Python version
+
+Before start using Fed-BioMed or start developing on Fed-BioMed please make sure that required Python version is installed on your machine. Please go to `pyrpoject.toml` file located in the root of Fed-BioMed clone, and check required Python version interval.
+
+!!! note "Tool for different Python version"
+
+    You can use the tool `pyenv` to be able to instal specific Python versions into your local environment
+    ```
+    pyenv install 3.11
+    pyenv global 3.11
+    ```
+    To code above will install latest Python 3.11, and activate this version globally.
+
+
+## Install recommended tools to manage development environment
+
+### Use a virtual environments to manage dependencies
+
+Using virtual environments will allow you to isolate your Fed-BioMed development environment from your other projects. There, you have of course several options. You can use `virtualenv` or `conda` or any other compatible Python virtual environment tools.
+
+While `conda` allows also installing a specified Python version, `virtualenv` requires that the Python, and Pip is already installed. Therefore, if `conda` is used using another tools like `pyenv` is not required to be installed and used to manage Python versions. However, if `virtualenv` is preferred, then, using tools `pyenv` to install required Python version is recommended.
+
+#### Case: venv
+
+The advantage of `venv` is that it keeps all project dependencies within your project folder and is part of `python` standard library. Please go to Fed-BioMed project root (Fed-BioMed) clone and execute following command.
+
+```
+cd <path-to-fedbiomed-clone>
+python -m venv dev-env
+```
+
+The command above will generate `dev-env` folder in the project folder where all the project dependencies are kept.
+
+To activate the environment you can execute following command from project root directory.
+
+```
+source dev-env/bin/activate
+```
+
+You can also use default naming convention `.venv` for virtual environment name.
+
+#### Case: Conda
+
```

Comment:
**here**: recommend the user to update his `conda` with the latest verision available

`conda update -n base-c conda-forge conda`